src/cascadia/TerminalApp/Tab.cpp

```diff
@@ -64,15 +85,87 @@ void Tab::SetFocused(bool focused)
     }
 }
 
-GUID Tab::GetProfile() const noexcept
+// Method Description:
+// - Returns nullopt if no children of this tab were the last control to be
+//   focused, or the GUID of the profile of the last control to be focused (if
+//   there was one).
+// Arguments:
+// - <none>
+// Return Value:
+// - nullopt if no children of this tab were the last control to be
+//   focused, else the GUID of the profile of the last control to be focused
+std::optional<GUID> Tab::GetFocusedProfile() const noexcept
+{
+    return _rootPane->GetFocusedProfile();
+}
+
+// Method Description:
+// - Attempts to update the settings of this tab's tree of panes.
+// Arguments:
+// - settings: The new TerminalSettings to apply to any matching controls
+// - profile: The GUID of the profile these settings should apply to.
+// Return Value:
+// - <none>
+void Tab::UpdateSettings(const TerminalSettings& settings, const GUID& profile)
 {
-    return _profile;
+    _rootPane->UpdateSettings(settings, profile);
 }
 
+// Method Description:
+// - Focus the last focused control in our tree of panes.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
 void Tab::_Focus()
 {
     _focused = true;
-    _control.GetControl().Focus(FocusState::Programmatic);
+
+    auto lastFocusedControl = _rootPane->GetFocusedTerminalControl();
+    if (lastFocusedControl)
+    {
+        lastFocusedControl.GetControl().Focus(FocusState::Programmatic);
+    }
+}
+
+// Method Description:
+// - Update the focus state of this tab's tree of panes. If one of the controls
+//   under this tab is focused, then it will be marked as the last focused. If
+//   there are no focused panes, then there will not be a last focused control
+//   when this returns.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void Tab::UpdateFocus()
+{
+    _rootPane->UpdateFocus();
+}
+
+// Method Description:
+// - Gets the title string of the last focused terminal control in our tree.
+//   Returns the empty string if there is no such control.
+// Arguments:
+// - <none>
+// Return Value:
+// - the title string of the last focused terminal control in our tree.
+winrt::hstring Tab::GetFocusedTitle() const
+{
+    const auto lastFocusedControl = _rootPane->GetFocusedTerminalControl();
+    return lastFocusedControl ? lastFocusedControl.Title() : L"";
+}
+
+// Method Description:
+// - Set the text on the TabViewItem for this tab.
+// Arguments:
+// - text: The new text string to use as the Header for our TabViewItem
+// Return Value:
+// - <none>
+void Tab::SetTabText(const winrt::hstring& text)
+{
+    _tabViewItem.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [=](){

```

Comment:
this might actually capture a dead reference to an hstring since it's not calling the copy ctor

Reply:
```suggestion
    // Copy the hstring, so we don't capture a dead reference
    winrt::hstring textCopy{ text };
    _tabViewItem.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [textCopy](){
```